fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/api/networking/v1/PlayerLookup.java

```diff
@@ -0,0 +1,193 @@
+	}
+
+	/**
+	 * Gets all players tracking an entity in a server world.
+	 *
+	 * <p>The returned collection is immutable.
+	 *
+	 * <p><b>Warning</b>: If the provided entity is a player, it is not
+	 * guaranteed by the contract that said player is included in the
+	 * resulting stream.
+	 *
+	 * @param entity the entity being tracked
+	 * @return the players tracking the entity
+	 * @throws IllegalArgumentException if the entity is not in a server world
+	 */
+	public static Collection<ServerPlayerEntity> tracking(Entity entity) {
+		Objects.requireNonNull(entity, "Entity cannot be null");
+		ChunkManager manager = entity.world.getChunkManager();
+
+		if (manager instanceof ServerChunkManager) {
+			ThreadedAnvilChunkStorage storage = ((ServerChunkManager) manager).threadedAnvilChunkStorage;
+
+			// return an immutable collection to guard against accidental removals.
+			return Collections.unmodifiableCollection(((ThreadedAnvilChunkStorageTrackingExtensions) storage).fabric_getTrackingPlayers(entity));
+		}
+
+		throw new IllegalArgumentException("Only supported on server worlds!");
+	}
+
+	/**
+	 * Gets all players tracking a block entity in a server world.
+	 *
+	 * @param blockEntity the block entity
+	 * @return the players tracking the block position
+	 * @throws IllegalArgumentException if the block entity is not in a server world
+	 */
+	public static Collection<ServerPlayerEntity> tracking(BlockEntity blockEntity) {
+		Objects.requireNonNull(blockEntity, "BlockEntity cannot be null");
+
+		//noinspection ConstantConditions - IJ intrinsics don't know hasWorld == true will result in no null
+		if (!blockEntity.hasWorld() || blockEntity.getWorld().isClient()) {
+			throw new IllegalArgumentException("Only supported on server worlds!");
+		}
+
+		return tracking((ServerWorld) blockEntity.getWorld(), blockEntity.getPos());
+	}
+
+	/**
+	 * Gets all players tracking a block position in a server world.
+	 *
+	 * @param world the server world
+	 * @param pos   the block position
+	 * @return the players tracking the block position
+	 */
+	public static Collection<ServerPlayerEntity> tracking(ServerWorld world, BlockPos pos) {
+		Objects.requireNonNull(pos, "BlockPos cannot be null");
+
+		return tracking(world, new ChunkPos(pos));
+	}
+
+	/**
+	 * Gets all players around a position in a world.
+	 *
+	 * <p>The distance check is done in the three-dimensional space instead of in the horizontal plane.
+	 *
+	 * @param world  the world
+	 * @param pos the position
+	 * @param radius the maximum distance from the position in blocks
+	 * @return the players around the position
+	 */
+	public static Collection<ServerPlayerEntity> around(ServerWorld world, Vec3d pos, double radius) {
+		double radiusSq = radius * radius;
+
+		return world(world)
+				.stream()
+				.filter((p) -> p.squaredDistanceTo(pos) <= radiusSq)
+				.collect(Collectors.toList());
```

Comment:
Would returning an `Iterable` or `Iterator` be sufficient in order to not creating a list?

Reply:
Having the size is quite nice and it needs to create something, the (Array)List should be one of the cheapest options.